src/servers/graphql-server.ts

```diff
@@ -135,32 +137,32 @@ const sessionContext = ({
     async () => {
       // note: value should match (ie: "anon") if not an accountId
       // settings from dev/ory/oathkeeper.yml/authenticator/anonymous/config/subjet
-      const maybeKratosUserId = checkedToKratosUserId(tokenPayload?.sub || "")
-      if (!(maybeKratosUserId instanceof ValidationError)) {
-        const userId = maybeKratosUserId
+      const maybeUserId = checkedToUserId(tokenPayload?.sub || "")
+      if (!(maybeUserId instanceof ValidationError)) {
+        const userId = maybeUserId
 
-        const account = await Accounts.getAccountFromKratosUserId(userId)
-        if (account instanceof Error) throw Error
+        const account = await Accounts.getAccountFromUserId(userId)
+        if (account instanceof Error) throw new Error(account.name)
         domainAccount = account
 
-        const loggedInUser = await Users.getUserForLogin({
-          userId: account.id as string as UserId,
+        // not awaiting on purpose. just updating metadata
+        Accounts.updateAccountIPsInfo({
+          accountId: account.id,
           ip,
           logger,
         })
-        if (loggedInUser instanceof Error)
-          throw new ApolloError("Invalid user authentication", "INVALID_AUTHENTICATION", {
-            reason: loggedInUser,
-          })
-        domainUser = loggedInUser
+
+        const userRes = await UsersRepository().findById(account.kratosUserId)
+        if (userRes instanceof Error) throw new Error(userRes.name)
```

Comment:
We should only be throwing `CustomApolloError` from here
```suggestion
        if (userRes instanceof Error) throw mapError(userRes)
```